examples/server/server.cpp

```diff
@@ -504,95 +500,207 @@ bool server_params_parse(int argc, char **argv, server_params &sparams, gpt_para
       fprintf(stderr, "warning: see main README.md for information on enabling GPU BLAS support\n");
 #endif
     }
+    else if (arg == "--lora")
+    {
+        if (++i >= argc)
+        {
+            invalid_param = true;
+            break;
+        }
+        params.lora_adapter = argv[i];
+        params.use_mmap = false;
+    }
+    else if (arg == "--lora-base")
+    {
+        if (++i >= argc) {
+            invalid_param = true;
+            break;
+        }
+        params.lora_base = argv[i];
+    } else if (arg == "-v" || arg == "--verbose") {
+        sparams.verbose = true;
+    }
     else
     {
       fprintf(stderr, "error: unknown argument: %s\n", arg.c_str());
-      server_print_usage(argc, argv, default_params);
+      server_print_usage(argc, argv, default_params, default_sparams);
       exit(1);
     }
   }
 
   if (invalid_param)
   {
     fprintf(stderr, "error: invalid parameter for argument: %s\n", arg.c_str());
-    server_print_usage(argc, argv, default_params);
+    server_print_usage(argc, argv, default_params, default_sparams);
     exit(1);
   }
   return true;
 }
 
-bool parse_options_completion(json body, llama_server_context& llama, Response &res) {
-  if (!body["threads"].is_null())
-  {
-    llama.params.n_threads = body["threads"].get<int>();
+json format_generation_settings(llama_server_context &llama) {
+  const bool ignore_eos = -INFINITY == llama.params.logit_bias[llama_token_eos()];
```

Comment:
```suggestion
  auto eos_bias = llama.params.logit_bias.find(llama_token_eos());
  const bool ignore_eos = eos_bias != llama.params.logit_bias.end() &&
                          eos_bias->second < 0.0f && std::isinf(eos_bias->second);

```

Reply:
`operator[]` will add it to the map if it's missing with the value 0. Also, checking infinty should not be done with `==`

Reply:
It may work but technically `isinf()` should be used since infinity does not equal anything.

I'll add the `const`

Reply:
I need to figure out how to pass -Infinity from the JSON